app/Http/Controllers/Licenses/LicensesController.php

```diff
@@ -248,16 +248,23 @@ public function show($licenseId = null)
         }
 
         $users_count = User::where('autoassign_licenses', '1')->count();
-        $total_seats_count = $license->totalSeatsByLicenseID();
+        $total_seats_count = (int) $license->totalSeatsByLicenseID();
         $available_seats_count = $license->availCount()->count();
-        $checkedout_seats_count = ($total_seats_count - $available_seats_count);
+        $unreassignable_seats_count = Helper::unReassignableCount($license);
+        if(!$license->reassignable){
+            $checkedout_seats_count = ($total_seats_count - $available_seats_count - $unreassignable_seats_count );
+        }
+        else {
+            $checkedout_seats_count = ($total_seats_count - $available_seats_count);
+        }
```

Comment:
This logic feels like it should live on the model since there's a good chance we'll want to use it elsewhere in the future. We can avoid copy/pasting by moving it there.

Reply:
moved to the LicenseSeat model, along with the `$total_seat_count` and `$available_seats_count`. returns an array for these values now.